snapcraft/plugins/python.py

```diff
@@ -163,24 +179,56 @@ def clean_pull(self):
         if os.path.isdir(self._python_package_dir):
             shutil.rmtree(self._python_package_dir)
 
+    def _get_python_command(self):
+        python_command = os.path.join(
+            'usr', 'bin', self.options.python_version)
+
+        # staged as in project.stage_dir, not from a stage-packages entry
+        # in this part.
+        staged_python = os.path.join(self.project.stage_dir, python_command)
+        unstaged_python = os.path.join(self.installdir, python_command)
+
+        if os.path.exists(staged_python):
+            return staged_python
+        else:
+            return unstaged_python
+
     def _install_pip(self, download):
         env = os.environ.copy()
         env['PYTHONUSERBASE'] = self.installdir
+        # since we are using an independent env we need to export this too
+        # TODO: figure out if we can move back to common.run
+        env['SNAPCRAFT_STAGE'] = self.project.stage_dir
+        env['SNAPCRAFT_PART_INSTALL'] = self.installdir
 
         args = ['pip', 'setuptools', 'wheel']
 
+        pip_command = [self._get_python_command(), '-m', 'pip']
+
+        # if python_command it is not from stage we don't have pip
```

Comment:
s/if python_command it is/if python_command is/. Although from the comment I'm not sure how that relates to the setting of `PYTHONHOME`.